climakitae/new_core/param_validation/convert_units_param_validator.py

```diff
@@ -0,0 +1,147 @@
+"""
+Validator for parameters provided to Convert Units Processor.
+"""
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, Iterable
+
+from climakitae.core.constants import UNSET
+from climakitae.new_core.param_validation.abc_param_validation import (
+    register_processor_validator,
+)
+
+# Valid unit options (from convert_units.py)
+UNIT_OPTIONS = {
+    "K": ["K", "degC", "degF"],
+    "degF": ["K", "degC", "degF"],
+    "degC": ["K", "degC", "degF"],
+    "hPa": ["Pa", "hPa", "mb", "inHg"],
+    "Pa": ["Pa", "hPa", "mb", "inHg"],
+    "m/s": ["m/s", "mph", "knots"],
+    "m s-1": ["m s-1", "mph", "knots"],
+    "[0 to 100]": ["[0 to 100]", "fraction"],
+    "mm": ["mm", "inches"],
+    "mm/d": ["mm/d", "inches/d"],
+    "mm/h": ["mm/h", "inches/h"],
+    "kg/kg": ["kg/kg", "g/kg"],
+    "kg kg-1": ["kg kg-1", "g kg-1"],
+    "kg m-2 s-1": ["kg m-2 s-1", "mm", "inches"],
+    "g/kg": ["g/kg", "kg/kg"],
+}
```

Comment:
This is duplicated code. Let's move it into a util file like `core.constants` or something.